packages/core/src/external/aws/lambda-logic/document-uploader.ts

```diff
@@ -0,0 +1,105 @@
+import axios from "axios";
+import { MetriportError } from "../../../util/error/metriport-error";
+import { S3Utils } from "../s3";
+
+const api = axios.create();
+
+export type FileData = {
+  mimeType?: string | undefined;
+  size?: number | undefined;
+  originalName: string;
+  locationUrl: string;
+  docId: string;
+  docRefId: string;
+};
+
+export async function documentUploaderHandler(
+  sourceBucket: string,
+  sourceKey: string,
+  destinationBucket: string,
+  region: string,
+  apiServerURL: string
+) {
+  const s3Utils = new S3Utils(region);
+  const copySource = encodeURI(`${sourceBucket}/${sourceKey}`);
+  const { destinationKey, docRefId } = getDestinationKeyAndDocRefId(sourceKey);
+
+  const params = {
+    CopySource: copySource,
+    Bucket: destinationBucket,
+    Key: destinationKey,
+  };
+
+  // Make a copy of the file to the general medical documents bucket
+  try {
+    await s3Utils.s3.copyObject(params).promise();
+  } catch (error) {
+    const message = "Error copying the uploaded file to medical documents bucket";
+    console.log(message, JSON.stringify(error));
+    throw new MetriportError(message, null, { copySource, destinationBucket, destinationKey });
+  }
+
+  // Get file info from the copied file
+  const { size, contentType } = await s3Utils.getFileInfoFromS3(destinationKey, destinationBucket);
+
+  const s3FileNameParts = s3Utils.parseS3FileName(destinationKey);
+  if (!s3FileNameParts) {
+    const message = "Invalid S3 file name";
+    throw new MetriportError(message, null, { destinationBucket, destinationKey });
+  }
+  const { cxId, docId } = s3FileNameParts;
+
+  const fileData: FileData = {
+    mimeType: contentType,
+    size,
+    originalName: destinationKey,
+    locationUrl: `https://${destinationBucket}.s3.${region}.amazonaws.com/${destinationKey}`,
```

Comment:
Since we have an instance of S3Utils, which is attached to a region, we could move this to a function there, something like `buildFileUrl(bucket, key)`